app/index.js

```diff
@@ -0,0 +1,213 @@
+const _ = require('lodash');
+const path = require('path');
+const bodyParser = require('body-parser');
+const express = require('express');
+const cookieParser = require('cookie-parser');
+const session = require('express-session');
+const knex = require('knex');
+const passport = require('passport');
+const LocalStrategy = require('passport-local').Strategy;
+const handlebars = require('express-handlebars');
+const flash = require('connect-flash');
+
+const ENV = process.env.NODE_ENV || 'development';
+
+const config = require('../knexfile');
+const db = knex(config[ENV]);
+
+// Initialize Express.
+const app = express();
+app.use(bodyParser.urlencoded({extended: false}));
+app.use(bodyParser.json());
+app.use(session({ secret: 'some secret' }));
+app.use(flash());
+app.use(cookieParser());
+app.use(passport.initialize());
+app.use(passport.session());
+
+app.engine('handlebars', handlebars({
+  defaultLayout: 'main',
+  layoutsDir: path.join(__dirname, '/views/layouts')
+}));
+app.set('views', path.join(__dirname, '/views'));
+app.set('view engine', 'handlebars');
+
+// Configure & Initialize Bookshelf & Knex.
+console.log(`Running in environment: ${ENV}`);
+
+// ***** Models ***** //
+
+const Comment = require('./models/comment');
+const Post = require('./models/post');
+const User = require('./models/user');
+
+/// ***** Passport Strategies & Helpers ***** //
+
+passport.use(new LocalStrategy((username, password, done) => {
+  User
+    .forge({ username: username })
+    .fetch()
+    .then((usr) => {
+      if (!usr) {
+        return done(null, false, { message: 'Incorrect username.' });
+      }
+      usr.validatePassword(password).then((valid) => {
+        if (!valid) {
+          return done(null, false, { message: 'Invalid password.' });
+        }
+        return done(null, usr);
+      });
+    })
+    .catch((err) => {
+      return done(err);
+    });
+}));
+
+passport.serializeUser(function(user, done) {
+  done(null, user.id);
+});
+
+passport.deserializeUser(function(user, done) {
```

Comment:
These functions are the worst parts of passport (I think, anyway), but I really like how you've explained and used them.